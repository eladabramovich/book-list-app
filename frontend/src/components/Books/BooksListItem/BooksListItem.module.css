.bookItem {
  width: 100%;
  margin-bottom: 4px;
}

.bookItem a {
  display: flex;
  width: 100%;
  text-decoration: none;
}

.bookItem a:hover,
.bookItem a:active {
  background-color: rgba(176, 196, 199, 0.11);
}

.bookItem img {
  margin-right: 16px;
}

.bookItem .bookDetails {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.bookItem .bookDetails h3,
.bookItem .bookDetails p {
  color: var(--text-color-primary);
  font-weight: 700;
}

.bookItem .bookDetails h3 {
  margin-bottom: 8px;
  font-size: 24px;
  line-height: 1.3;
}

.bookItem .bookDetails p {
  font-size: 13px;
}

.divider {
  margin-bottom: 16px;
  border: 1px solid #e7edf3;
}

@media screen and (max-width: 767px) {
  .bookItem a {
    flex-direction: column;
  }

  .bookItem picture {
    display: flex;
    justify-content: center;
  }

  .bookItem img {
    margin-bottom: 16px;
    object-fit: contain;
  }

  .bookItem .bookDetails {
    align-items: center;
  }

  .bookItem .bookDetails h3,
  .bookItem .bookDetails p {
    text-align: center;
  }

  .bookItem .bookDetails p {
    margin-bottom: 8px;
  }
}
