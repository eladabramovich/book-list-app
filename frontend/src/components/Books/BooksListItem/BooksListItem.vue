<template>
  <div>
    <div :class="styles.bookItem">
      <router-link :to="`/book/${id}`">
        <picture>
          <source
            media="(min-width: 769px)"
            :srcset="images.small"
            type="image/jpeg"
          />
          <source
            media="(max-width: 768px)"
            :srcset="images.large"
            type="image/jpeg"
          />
          <img :src="images.small" :alt="title" />
        </picture>
        <div :class="styles.bookDetails">
          <h3>{{ title }}</h3>
          <p>Score: {{ formattedScore }}</p>
        </div>
      </router-link>
    </div>
    <hr :class="styles.divider" />
  </div>
</template>

<script>
import moduleStyles from './BooksListItem.module.css';
export default {
  props: {
    id: {
      type: String,
      required: true,
    },
    title: {
      type: String,
      required: true,
    },
    images: {
      type: Object,
      required: true,
    },
    score: {
      type: Number,
      default: 0.0,
    },
  },
  computed: {
    styles() {
      return moduleStyles;
    },
    formattedScore() {
      return Number.isInteger(this.score) ? this.score : this.score.toFixed(2);
    },
  },
};
</script>
