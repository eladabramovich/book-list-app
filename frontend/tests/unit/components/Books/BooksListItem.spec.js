import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import BooksListItem from '@/components/Books/BooksListItem/BooksListItem.vue';

describe('BooksListItem.vue', () => {
  it('Should render', () => {
    const wrapper = shallowMount(BooksListItem, {
      stubs: {
        RouterLink: RouterLinkStub,
      },
      propsData: {
        id: '1',
        title: 'Moby Dick',
        images: {
          small: 'http://localhost:5000/images/Moby_Dick_96x132.jpg',
          large: 'http://localhost:5000/images/Moby_Dick_255x387.jpg',
        },
        score: 0.0,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
