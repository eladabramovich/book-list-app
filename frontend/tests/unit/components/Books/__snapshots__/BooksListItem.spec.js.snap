// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksListItem.vue Should render 1`] = `
<div>
  <div class="bookItem"><a>
      <picture>
        <source media="(min-width: 769px)" srcset="http://localhost:5000/images/Moby_Dick_96x132.jpg" type="image/jpeg">
        <source media="(max-width: 768px)" srcset="http://localhost:5000/images/Moby_Dick_255x387.jpg" type="image/jpeg"> <img src="http://localhost:5000/images/Moby_Dick_96x132.jpg" alt="Moby Dick">
      </picture>
      <div class="bookDetails">
        <h3>Moby Dick</h3>
        <p>Score: 0</p>
      </div>
    </a></div>
  <hr class="divider">
</div>
`;
